src/create_disk.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/sh
```

Comment:
This is minor, but how about we `set -euo pipefail` right from the start?
Seems like the only change is just doing a `if [ $# -ne 6 ]; then usage; fi` and then you can `shift` in peace.

Especially when debugging stuff like this running inside a VM running inside a container, `set -u` is super useful.